OUDS/Foundations/Sources/iPhoneDevice.swift

```diff
@@ -0,0 +1,225 @@
+//
+// Software Name: OUDS iOS
+// SPDX-FileCopyrightText: Copyright (c) Orange SA
+// SPDX-License-Identifier: MIT
+//
+// This software is distributed under the MIT license,
+// the text of which is available at https://opensource.org/license/MIT/
+// or see the "LICENSE" file for more details.
+//
+// Authors: See CONTRIBUTORS.txt
+// Software description: A SwiftUI components library with code examples for Orange Unified Design System
+//
+
+import Foundation
+import SwiftUI
+
+// swiftlint:disable type_name
+// swiftlint:disable line_length
+// swiftlint:disable switch_case_on_newline
+
+// MARK: - Environment
+
+private struct iPhoneDeviceEnvironmentKey: EnvironmentKey {
+
+    static let defaultValue = iPhoneDevice.unknown
+}
+
+extension EnvironmentValues {
+
+    /// The `iPhoneDevice` in use, defined depending to screen dimensions.
+    /// No need to get more details (at least for privacy concerns), just to know why group of devices it is part of.
+    /// Can be useful to compute some lements like indicators of tab bars.
+    public var iPhoneInUse: iPhoneDevice {
+        get {
+            self[iPhoneDeviceEnvironmentKey.self]
+        }
+        set {
+            self[iPhoneDeviceEnvironmentKey.self] = newValue
+        }
+    }
+}
+
+// MARK: - iPhone Device
+
+/// List some iPhone devices which can use OUDS, i.e. supposed to support at least iOS 15.
+/// Helps to define some metrics like tab bar heights.
+///
+/// - Since: 1.0.0
+public enum iPhoneDevice: CustomStringConvertible, Sendable {
```

Comment:
used to get tabbarheight ?

why don't use the same way as setupTabBarAppearance using UItabbar to get the height and the number of items and set everything in a view modifier that will set Indicator on tab bar. ?

Reply:
Je n'ai rien compris.
Tu suggères de récupérer _by code_ le nombre de _tab bar item_ et de voir ainsi quelle serait leur hauteur ?